roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,354 @@
+        """Save the first and last 500 lines of logs for the specified
+        component and container."""
+        out = self.read_command_output(' '.join(['/usr/local/bin/master-logs',
+                                                 component, container, '2<&1',
+                                                 '|', '/bin/awk',
+                                                 """'
+                                                 NR <= 500 {print}
+                                                 NR > 500 {buf[(NR-1)%500+1]=$0}
+                                                 END {
+                                                   if (NR>1000) print "...";
+                                                   for (i=0; i < 500; ++i)
+                                                     print buf[(NR+i)%500+1]
+                                                 }
+                                                 '"""]))
+        self.register_file('master-logs_%s_%s' % (component, container), out)
+
+    def get_resource(self, kind):
+        """Return a list of all resources of the specified kind."""
+        for resource in self.task_vars['resources']['results']:
+            if resource['item'] == kind:
+                return resource['results']['results'][0]['items']
+
+        raise OpenShiftCheckException('CouldNotListResource',
+                                      'Could not list resource %s' % kind)
+
+    def check_node(self):
+        """Gather diagnostic information on a node and perform connectivity
+        checks on pods and services."""
+        node_name = self.get_var('openshift', 'node', 'nodename', default=None)
+        if not node_name:
+            self.register_failure('Could not determine node name.')
+            return
+
+        # The "openvswitch" container uses the host netnamespace, but the host
+        # file system may not have the brctl, ovs-appctl, and ovs-ofctl
+        # binaries, which we use for some diagnostics.  Thus we run these
+        # binaries inside the container, and to that end, we need to determine
+        # its container id.
+        ovs_cid = self.read_command_output([
+            '/bin/docker', 'ps', '-l', '-a', '-q',
+            '--filter=label=io.kubernetes.container.name=openvswitch',
+            '--filter=label=io.kubernetes.pod.namespace=openshift-sdn'
+        ])
+
+        if self.want_full_results:
+            try:
+                service_prefix = self.get_var('openshift_service_type')
+                if self._templar is not None:
+                    service_prefix = self._templar.template(service_prefix)
+                self.save_service_logs('%s-node' % service_prefix)
+
+                self.save_command_output('bridges',
+                                         ['/bin/docker', 'exec', ovs_cid,
+                                          '/sbin/brctl', 'show'])
+
+                self.save_command_output('docker-unit-file',
```

Comment:
Next 4 commands are specific to docker runtime. We will be using crio as default runtime, so we should detect the runtime or get it from the node config and generate the commands for capturing this output based on the runtime.

Reply:
Fixed.